### ISSUE TITLE
Expired token not refreshed when leaving browser window open

### ISSUE THREAD

**Issue by fikkatra** [AzureAD/azure-activedirectory-library-for-js#593]
When leaving the app open until the token expires and then trying to do something with it, nothing happens. Adal seems not to be able to refresh the token. I'm running adal-angular v1.0.14.

Scenario:

- open app
- log in => everything works fine
- leave app open in the browser for more than 1 hour
- do something that triggers a (secured) API request
- API request does not go out

Additional debugging information:

- An authorize request goes out, but apparently doesn't do much (no new token is stored). A second authorize request is then fired, but is canceled by Chrome:

![image](https://user-images.githubusercontent.com/8104336/28312158-71999e36-6bb2-11e7-945e-4f41cc4b81af.png)

- The event handler of the adal:loginFailure event is not hit
- Refreshing doesn't work. The only thing that works, is re-open the app in another browser tab, I guess because this starts a new session.

This is my configuration:
```
adalAuthenticationServiceProvider.init(
	{
		instance: 'https://login.microsoftonline.com/',
		tenant: '<azure ad tenant>',
		clientId: '<azure ad client id>',
		endpoints: {
			"/api": '<azure ad client id>'
		},
		anonymousEndpoints: ["locales"]
	},
	$httpProvider
)
```
I'm using ui router and all my routes have the requireADLogin set to true.

Any solutions or workarounds?

**Comment by jmprieur** [comment]
Thanks for your feedback @fikkatra 
Could you please try with ADAL.js 1.0.15 ?

**Comment by fikkatra** [comment]
@jmprieur I can confirm that the problem is fixed in v1.0.15. Thanks!

**Comment by jmprieur** [comment]
Thanks for confirming, @fikkatra
